src/phpDocumentor/Reflection/Php/Factory/Class_.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+
+namespace phpDocumentor\Reflection\Php\Factory;
+
+use InvalidArgumentException;
+use phpDocumentor\Descriptor\Class_ as ClassDescriptor;
+use phpDocumentor\Descriptor\Constant;
+use phpDocumentor\Descriptor\Method as MethodDescriptor;
+use phpDocumentor\Descriptor\Property as PropertyDescriptor;
+use phpDocumentor\Reflection\Fqsen;
+use phpDocumentor\Reflection\Php\ProjectFactoryStrategy;
+use phpDocumentor\Reflection\Php\StrategyContainer;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_ as ClassNode;
+use PhpParser\Node\Stmt\ClassConst;
+use PhpParser\Node\Stmt\Property as PropertyNode;
+use PhpParser\Comment\Doc;
+use PhpParser\Node\Stmt\TraitUse;
+
+/**
+ * Strategy to create a ClassDescriptor including all sub elements.
+ */
+final class Class_ implements ProjectFactoryStrategy
+{
+
+    /**
+     * Returns true when the strategy is able to handle the object.
+     *
+     * @param object $object object to check.
+     * @return boolean
+     */
+    public function matches($object)
+    {
+        return $object instanceof ClassNode;
+    }
+
+    /**
+     * Creates an ClassDescriptor out of the given object.
+     * Since an object might contain other objects that need to be converted the $factory is passed so it can be
+     * used to create nested Elements.
+     *
+     * @param ClassNode $object object to convert to an Element
+     * @param StrategyContainer $strategies used to convert nested objects.
+     * @return ClassDescriptor
+     */
+    public function create($object, StrategyContainer $strategies)
```

Comment:
This method is too large, is it possible to divide it into more specific private methods?